app/models/estimated_population.rb

```diff
@@ -0,0 +1,48 @@
+class EstimatedPopulation < ApplicationRecord
+  belongs_to :region
+
+  validates :population, presence: true, numericality: true
+  validates :diagnosis, presence: true
+
+  enum diagnosis: {HTN: "HTN", DM: "DM"}
+
+  validate :can_only_be_set_for_district_or_state
+  after_commit :update_state_population
+
+  def can_only_be_set_for_district_or_state
+    region_type = region.region_type
+
+    unless region_type == "district" || region_type == "state"
```

Comment:
```suggestion
    unless region.district_region? || region.state_region?
```

then the `region_type` local can go away